bundle/edu.gemini.spdb.reports.collection/src/main/scala/edu/gemini/dbTools/mailer/Mailer.scala

```diff
@@ -0,0 +1,131 @@
+package edu.gemini.dbTools.mailer
+
+
+import edu.gemini.dbTools.mailer.MailerType._
+import edu.gemini.spModel.core.Site
+
+import java.util.logging.{ Logger, Level }
+
+import javax.mail._
+import javax.mail.Message.RecipientType.TO
+import javax.mail.internet._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+import scalaz._
+import Scalaz._
+
+import scalaz.effect.IO
+import scalaz.effect.IO._
+
+
+sealed abstract class Mailer(val site: Site, val smtpHost: String) {
+
+  def sendText(
+    to:      List[InternetAddress],
+    subject: String,
+    message: String
+  ): IO[Unit] = IO {
+
+    val sender = new InternetAddress       <|
+      (_.setAddress("noreply@gemini.edu")) <|
+      (_.setPersonal(s"Gemini ODB (${site.abbreviation})"))
+
+    val props = new java.util.Properties         <|
+      (_.put("mail.transport.protocol", "smtp")) <|
+      (_.put("mail.smtp.host", smtpHost))
+
+    new MimeMessage(Session.getInstance(props, null)) <|
+      (_.setFrom(sender))                             <|
+      (m => to.foreach(m.addRecipient(TO, _)))        <|
+      (_.setSubject(subject))                         <|
+      (_.setContent(message, "text/plain"))
+
+  } >>= send
+
+  def send(m: MimeMessage): IO[Unit]
+
+}
+
+object Mailer {
+
+  val Log = Logger.getLogger(getClass.getName)
+
+  private def sendAsync(m: MimeMessage): Unit = {
+    // OCSADV-85: use a future here; it can block for a long time if the mail server is down
+    val to = m.getRecipients(TO).mkString(", ")
+    Future(Transport.send(m)).onComplete {
+      case Success(_) => Log.info(s"Successfully sent mail to $to.")
+      case Failure(e) => Log.log(Level.WARNING, s"Failed to send mail to $to", e)
+    }
+  }
+
+  private def write(m: MimeMessage): IO[Unit] = {
+    def log(m: String): IO[Unit] =
+      putStrLn(s"TestMailer: $m")
+
+    for {
+      _ <- putStrLn("")
+      _ <- log(s"From: ${m.getFrom.mkString(", ")}")
+      _ <- log(s"To: ${m.getRecipients(TO).mkString(", ")}")
+      _ <- log(s"Subject: ${m.getSubject}")
+      _ <- log("")
+      _ <- m.getContent.toString.lines.toList.traverseU(log)
+    } yield ()
+  }
+
+  def apply(site: Site, smtpHost: String): Mailer =
+    new Mailer(site, smtpHost) {
+      def send(m: MimeMessage): IO[Unit] =
+        IO(sendAsync(m))
+    }
+
+  def forTesting(site: Site, smtpHost: String): Mailer =
+    new Mailer(site, smtpHost) {
+      private def isGeminiAddress(a: Address): Boolean =
+        a match {
+          case ia: InternetAddress => Option(ia.getAddress).exists(_.trim.endsWith("@gemini.edu"))
+          case _                   => false
+        }
+
+      def send(m: MimeMessage): IO[Unit] = {
+        write(m) *> {
+
+          val allRecipients = m.getRecipients(TO)
+
+          val content       =
```

Comment:
Is there any need not to hardcode the messages and read them from config instead?

Reply:
Here it is just wrapping whatever the actual message is in some text that explains that this is just a test.

The REL task didn't mention any specific text for the production version of the email itself so I invented something for a first pass.  I expect they'll want to modify that text, but that it won't change so much that we need to go to the trouble of putting it in a config file.